metals/src/main/scala/scala/meta/internal/metals/codelenses/RunTestCodeLens.scala

```diff
@@ -57,25 +63,89 @@ final class RunTestCodeLens(
       Seq.empty
     } else {
       val distance = buffers.tokenEditDistance(path, textDocument.text, trees)
-
       val lenses = for {
         buildTargetId <- buildTargets.inverseSources(path)
         buildTarget <- buildTargets.info(buildTargetId)
         if buildTarget.getCapabilities.getCanDebug || isBloopOrSbt(),
       } yield {
         val classes = buildTargetClasses.classesOf(buildTargetId)
-        codeLenses(textDocument, buildTargetId, classes, distance)
+        codeLenses(textDocument, buildTargetId, classes, distance, path)
       }
 
       lenses.getOrElse(Seq.empty)
     }
   }
 
+  private def isMainMethod(signature: Signature, textDocument: TextDocument) = {
+    signature match {
+      case MethodSignature(_, parameterLists, returnType) =>
+        def isVoid = returnType match {
+          case TypeRef(_, symbol, _) => symbol == "scala/Unit#"
+          case _ => false
+        }
+
+        def isStringArray(sym: String) = {
+          textDocument.symbols.find(
+            _.symbol == sym
+          ) match {
+            case Some(info) =>
+              info.signature match {
+                case ValueSignature(
+                      TypeRef(
+                        _,
+                        "scala/Array#",
+                        Vector(TypeRef(_, "java/lang/String#", _))
+                      )
+                    ) =>
+                  true
+              }
+            case None => false
+          }
+        }
+
+        isVoid &&
+        parameterLists.size == 1 &&
+        parameterLists.head.symlinks.size == 1 &&
+        isStringArray(parameterLists.head.symlinks.head)
+      case _ => false
+    }
+  }
+
+  private def javaLenses(
+      occurence: SymbolOccurrence,
+      textDocument: TextDocument,
+      target: BuildTargetIdentifier
+  ): Seq[l.Command] = {
+
+    if (occurence.symbol.endsWith("#main().")) {
+      textDocument.symbols.find(_.symbol == occurence.symbol).toSeq.flatMap {
+        a =>
```

Comment:
could it be formated like for example:
```suggestion
      textDocument.symbols.find(_.symbol == occurence.symbol).toSeq
      .flatMap { a =>
```
I mean, I just want `a =>` in the same line as the `flatMap` is, otherwise it looks strange to me.